File: C/readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int count_letters(string gradetext);
int count_words(string gradetext);
int count_sentences(string gradetext);

int main(void)
{
    string gradetext = get_string("Text: ");
    float letters = count_letters(gradetext);
    float words = count_words(gradetext);
    float sentences = count_sentences(gradetext);

    // L = avrage letters in 100 words
    // S = avrage sentencces in 100 words
    float L = 100 * letters / words;
    float S = 100 * sentences / words;

    float index = 0.0588 * L - 0.296 * S - 15.8;

    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 15)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", (int) round(index));
    }
    // printf("Letters: %i \n", letters);
    // printf("Words: %i \n", words);
    // printf("Sentences: %i \n", sentences);
}

int count_letters(string gradetext)
{
    float letters = 0;
    for (int i = 0, lenth = strlen(gradetext); i < lenth; i++)
    {
        // if upper or lower char letters + 1
        if (isupper(gradetext[i]) || islower(gradetext[i]))
        {
            letters++;
        }
    }
    return letters;
}
int count_words(string gradetext)
{
    float words = 0;
    int lenth = strlen(gradetext);

    if (lenth > 0)
    {
        words++;
    }
    for (int i = 0; i < lenth; i++)
    {
        // if upper or lower char letters + 1
        if (isspace(gradetext[i]))
        {
            words++;
        }
    }
    return words;
}
int count_sentences(string gradetext)
{
    float sentences = 0;

    for (int i = 0, lenth = strlen(gradetext); i < lenth; i++)
    {
        // if upper or lower char letters + 1
        if ((gradetext[i] == '.') || (gradetext[i] == '?') || (gradetext[i] == '!'))
        {
            sentences++;
        }
    }
    return sentences;
}

File: Python/sentimental-readability/readability.py
from cs50 import get_string
from cs50 import get_float


def main():
    gradetext = get_string("Input text to be graded: ")
    letters = countletters(gradetext)
    words = countwords(gradetext)
    sentences = countsentences(gradetext)

    let = 100 * (letters / words)
    sen = 100 * (sentences / words)

    index = 0.0588 * let - 0.296 * sen - 15.8
    index = round(index)
    if index < 1:
        print("Before Grade 1")
    elif index > 15:
        print("Grade 16+")
    else:
        print("Grade " + str(index))


def countletters(gradetext):
    letters = 0
    for i in range(len(gradetext)):
        if gradetext[i].isupper() or gradetext[i].islower():
            letters += 1
    return letters


def countwords(gradetext):
    words = len(gradetext.split())
    return words


def countsentences(gradetext):
    sentences = 0
    for i in range(len(gradetext)):
        if gradetext[i] == "." or gradetext[i] == "?" or gradetext[i] == "!":
            sentences += 1
    return sentences


if __name__ == "__main__":
    main()
